### PR TITLE
Remove iOS 15 support

### PR DESCRIPTION
**Pull request by mischreiber**
### Platforms Impacted
- [x] iOS
- [ ] visionOS
- [ ] macOS

### Description of changes

Updating iOS library to require iOS 16 as a minimum.

Removing some deprecated code as part of this, including:
* Simplifying how `FluentList` overrides its background color, using the new iOS 16 API `. scrollContentBackground(.hidden)`.
* Removing legacy swizzling code for handling content insets in `UIHostingController`-backed UI. We can now use the public API `sizingOptions = [.intrinsicContentSize]`.

### Binary change

Total increase: 70,120 bytes
Total decrease: -104,016 bytes
| File | Before | After | Delta |
|------|-------:|------:|------:|
| Total | 31,206,232 bytes | 31,172,336 bytes | 🎉 -33,896 bytes |
<details>
<summary> Full breakdown </summary>

| File | Before | After | Delta |
|------|-------:|------:|------:|
| PeoplePicker.o | 296,080 bytes | 332,424 bytes | ⚠️ 36,344 bytes |
| ActivityIndicatorCell.o | 75,464 bytes | 80,312 bytes | ⚠️ 4,848 bytes |
| ControlHostingView.o | 37,768 bytes | 42,096 bytes | ⚠️ 4,328 bytes |
| SearchBar.o | 368,952 bytes | 373,032 bytes | ⚠️ 4,080 bytes |
| MSFPersonaButton.o | 41,256 bytes | 45,176 bytes | ⚠️ 3,920 bytes |
| FluentList.o | 73,544 bytes | 76,104 bytes | ⚠️ 2,560 bytes |
| ListItem.o | 327,232 bytes | 328,552 bytes | ⚠️ 1,320 bytes |
| CardView.o | 243,000 bytes | 243,736 bytes | ⚠️ 736 bytes |
| UIColor+Extensions.o | 61,832 bytes | 62,552 bytes | ⚠️ 720 bytes |
| ShyHeaderView.o | 119,576 bytes | 120,192 bytes | ⚠️ 616 bytes |
| TabBarView.o | 141,680 bytes | 142,280 bytes | ⚠️ 600 bytes |
| Tooltip.o | 132,816 bytes | 133,304 bytes | ⚠️ 488 bytes |
| NavigationAnimator.o | 117,168 bytes | 117,624 bytes | ⚠️ 456 bytes |
| DrawerShadowView.o | 73,816 bytes | 74,256 bytes | ⚠️ 440 bytes |
| DateTimePicker.o | 168,752 bytes | 169,128 bytes | ⚠️ 376 bytes |
| NSLayoutConstraint+Extensions.o | 27,512 bytes | 27,864 bytes | ⚠️ 352 bytes |
| CommandBarCommandGroupsView.o | 205,912 bytes | 206,256 bytes | ⚠️ 344 bytes |
| CardNudge.o | 437,168 bytes | 437,504 bytes | ⚠️ 336 bytes |
| PersonaButtonCarousel.o | 168,008 bytes | 168,312 bytes | ⚠️ 304 bytes |
| MultilineCommandBar.o | 139,352 bytes | 139,608 bytes | ⚠️ 256 bytes |
| CenteredLabelCell.o | 83,160 bytes | 83,400 bytes | ⚠️ 240 bytes |
| ActivityIndicator.o | 158,672 bytes | 158,912 bytes | ⚠️ 240 bytes |
| SideTabBar.o | 206,952 bytes | 207,192 bytes | ⚠️ 240 bytes |
| HeadsUpDisplay.o | 253,280 bytes | 253,512 bytes | ⚠️ 232 bytes |
| IndeterminateProgressBar.o | 230,616 bytes | 230,840 bytes | ⚠️ 224 bytes |
| CommandBarButtonGroupView.o | 59,848 bytes | 60,072 bytes | ⚠️ 224 bytes |
| SegmentPillButton.o | 88,616 bytes | 88,824 bytes | ⚠️ 208 bytes |
| CalendarViewWeekdayHeadingView.o | 83,480 bytes | 83,688 bytes | ⚠️ 208 bytes |
| PersonaButton.o | 252,952 bytes | 253,160 bytes | ⚠️ 208 bytes |
| PopupMenuItemCell.o | 162,376 bytes | 162,552 bytes | ⚠️ 176 bytes |
| ActionsCell.o | 160,720 bytes | 160,888 bytes | ⚠️ 168 bytes |
| DotView.o | 32,192 bytes | 32,360 bytes | ⚠️ 168 bytes |
| DateTimePickerController.o | 143,144 bytes | 143,304 bytes | ⚠️ 160 bytes |
| TabBarItem.o | 68,632 bytes | 68,792 bytes | ⚠️ 160 bytes |
| CardPresentationController.o | 47,072 bytes | 47,224 bytes | ⚠️ 152 bytes |
| PageCardPresenterController.o | 149,296 bytes | 149,448 bytes | ⚠️ 152 bytes |
| CommandBarItem.o | 105,896 bytes | 106,048 bytes | ⚠️ 152 bytes |
| FluentUIFramework.o | 78,520 bytes | 78,672 bytes | ⚠️ 152 bytes |
| FontInfo.o | 73,768 bytes | 73,912 bytes | ⚠️ 144 bytes |
| ShimmerLinesView.o | 109,848 bytes | 109,992 bytes | ⚠️ 144 bytes |
| TooltipViewController.o | 46,248 bytes | 46,392 bytes | ⚠️ 144 bytes |
| BadgeLabelButton.o | 110,408 bytes | 110,544 bytes | ⚠️ 136 bytes |
| CommandingItem.o | 92,488 bytes | 92,624 bytes | ⚠️ 136 bytes |
| PillButtonBar.o | 248,688 bytes | 248,816 bytes | ⚠️ 128 bytes |
| PillButton.o | 162,184 bytes | 162,312 bytes | ⚠️ 128 bytes |
| NavigationController.o | 157,976 bytes | 158,096 bytes | ⚠️ 120 bytes |
| DatePickerController.o | 327,536 bytes | 327,648 bytes | ⚠️ 112 bytes |
| PersonaCell.o | 62,096 bytes | 62,208 bytes | ⚠️ 112 bytes |
| CalendarViewLayout.o | 73,224 bytes | 73,320 bytes | ⚠️ 96 bytes |
| FluentTheme.o | 212,248 bytes | 212,344 bytes | ⚠️ 96 bytes |
| BadgeLabel.o | 59,824 bytes | 59,920 bytes | ⚠️ 96 bytes |
| ListActionItem.o | 222,592 bytes | 222,688 bytes | ⚠️ 96 bytes |
| Separator.o | 70,128 bytes | 70,224 bytes | ⚠️ 96 bytes |
| BooleanCell.o | 83,640 bytes | 83,736 bytes | ⚠️ 96 bytes |
| CalendarViewDataSource.o | 128,120 bytes | 128,208 bytes | ⚠️ 88 bytes |
| ResizingHandleView.o | 69,120 bytes | 69,208 bytes | ⚠️ 88 bytes |
| CommandBarButton.o | 92,384 bytes | 92,472 bytes | ⚠️ 88 bytes |
| ColorProviding.o | 45,088 bytes | 45,176 bytes | ⚠️ 88 bytes |
| ShadowInfo.o | 48,464 bytes | 48,552 bytes | ⚠️ 88 bytes |
| CalendarViewDayMonthCell.o | 43,072 bytes | 43,152 bytes | ⚠️ 80 bytes |
| CalendarView.o | 65,680 bytes | 65,760 bytes | ⚠️ 80 bytes |
| BadgeStringExtractor.o | 65,048 bytes | 65,120 bytes | ⚠️ 72 bytes |
| DateTimePickerViewComponentTableView.o | 68,296 bytes | 68,368 bytes | ⚠️ 72 bytes |
| FluentButtonStyle.o | 146,280 bytes | 146,344 bytes | ⚠️ 64 bytes |
| DrawerTransitionAnimator.o | 72,576 bytes | 72,640 bytes | ⚠️ 64 bytes |
| PersonaListView.o | 183,128 bytes | 183,184 bytes | ⚠️ 56 bytes |
| FluentTextFieldInternal.o | 48,992 bytes | 49,048 bytes | ⚠️ 56 bytes |
| DateTimePickerViewComponent.o | 139,112 bytes | 139,160 bytes | ⚠️ 48 bytes |
| BarButtonItems.o | 33,840 bytes | 33,880 bytes | ⚠️ 40 bytes |
| DimmingView.o | 52,520 bytes | 52,560 bytes | ⚠️ 40 bytes |
| TwoLineTitleView+Navigation.o | 29,048 bytes | 29,088 bytes | ⚠️ 40 bytes |
| CalendarViewDayMonthYearCell.o | 42,120 bytes | 42,160 bytes | ⚠️ 40 bytes |
| TouchForwardingView.o | 33,952 bytes | 33,992 bytes | ⚠️ 40 bytes |
| Persona.o | 76,904 bytes | 76,936 bytes | ⚠️ 32 bytes |
| CardTransitionAnimator.o | 64,968 bytes | 65,000 bytes | ⚠️ 32 bytes |
| Button.o | 198,504 bytes | 198,528 bytes | ⚠️ 24 bytes |
| DateTimePickerViewComponentCell.o | 58,712 bytes | 58,736 bytes | ⚠️ 24 bytes |
| UIBarButtonItem+BadgeValue.o | 31,368 bytes | 31,376 bytes | ⚠️ 8 bytes |
| AvatarTitleView.o | 182,976 bytes | 182,968 bytes | 🎉 -8 bytes |
| AvatarGroup.o | 383,496 bytes | 383,488 bytes | 🎉 -8 bytes |
| DrawerPresentationController.o | 237,104 bytes | 237,088 bytes | 🎉 -16 bytes |
| PopupMenuItem.o | 127,544 bytes | 127,528 bytes | 🎉 -16 bytes |
| TabBarItemView.o | 219,440 bytes | 219,408 bytes | 🎉 -32 bytes |
| MSFActivityIndicator.o | 32,648 bytes | 32,608 bytes | 🎉 -40 bytes |
| MSFIndeterminateProgressBar.o | 30,096 bytes | 30,048 bytes | 🎉 -48 bytes |
| CardNudgeModifiers.o | 45,640 bytes | 45,592 bytes | 🎉 -48 bytes |
| SwiftUI+ViewModifiers.o | 131,504 bytes | 131,448 bytes | 🎉 -56 bytes |
| DateTimePickerViewLayout.o | 58,264 bytes | 58,200 bytes | 🎉 -64 bytes |
| FluentTextField.o | 192,080 bytes | 192,016 bytes | 🎉 -64 bytes |
| UINavigationItem+Navigation.o | 77,088 bytes | 77,024 bytes | 🎉 -64 bytes |
| TwoLineTitleViewTokenSet.o | 56,672 bytes | 56,608 bytes | 🎉 -64 bytes |
| UIView+Extensions.o | 45,824 bytes | 45,752 bytes | 🎉 -72 bytes |
| FluentTheme+UIKit.o | 27,024 bytes | 26,952 bytes | 🎉 -72 bytes |
| BlurringView.o | 35,440 bytes | 35,368 bytes | 🎉 -72 bytes |
| DateTimePickerViewDataSource.o | 279,416 bytes | 279,344 bytes | 🎉 -72 bytes |
| AvatarTitleViewTokenSet.o | 51,680 bytes | 51,608 bytes | 🎉 -72 bytes |
| FluentNotification.o | 644,848 bytes | 644,768 bytes | 🎉 -80 bytes |
| BadgeLabelTokenSet.o | 53,904 bytes | 53,824 bytes | 🎉 -80 bytes |
| SegmentItem.o | 33,360 bytes | 33,280 bytes | 🎉 -80 bytes |
| FluentListModifiers.o | 13,568 bytes | 13,480 bytes | 🎉 -88 bytes |
| AvatarModifiers.o | 46,040 bytes | 45,952 bytes | 🎉 -88 bytes |
| AnimationSynchronizer.o | 31,560 bytes | 31,472 bytes | 🎉 -88 bytes |
| CalendarViewDayTodayCell.o | 37,640 bytes | 37,552 bytes | 🎉 -88 bytes |
| ControlTokenSet.o | 76,592 bytes | 76,504 bytes | 🎉 -88 bytes |
| CalendarConfiguration.o | 49,120 bytes | 49,024 bytes | 🎉 -96 bytes |
| DrawerTokenSet.o | 54,904 bytes | 54,808 bytes | 🎉 -96 bytes |
| HUDModifiers.o | 65,568 bytes | 65,472 bytes | 🎉 -96 bytes |
| NotificationModifiers.o | 72,448 bytes | 72,336 bytes | 🎉 -112 bytes |
| BottomSheetTokenSet.o | 48,248 bytes | 48,136 bytes | 🎉 -112 bytes |
| ActivityIndicatorTokenSet.o | 63,696 bytes | 63,576 bytes | 🎉 -120 bytes |
| HeadsUpDisplayTokenSet.o | 56,976 bytes | 56,856 bytes | 🎉 -120 bytes |
| BottomSheetPassthroughView.o | 22,056 bytes | 21,936 bytes | 🎉 -120 bytes |
| SearchBarTokenSet.o | 82,904 bytes | 82,776 bytes | 🎉 -128 bytes |
| AccessibilityContainerView.o | 37,440 bytes | 37,312 bytes | 🎉 -128 bytes |
| CardPresenterNavigationController.o | 27,920 bytes | 27,792 bytes | 🎉 -128 bytes |
| TableViewCellTokenSet.o | 130,232 bytes | 130,104 bytes | 🎉 -128 bytes |
| EasyTapButton.o | 28,176 bytes | 28,048 bytes | 🎉 -128 bytes |
| LinearGradientInfo.o | 47,320 bytes | 47,184 bytes | 🎉 -136 bytes |
| FluentButtonToggleStyle.o | 65,656 bytes | 65,512 bytes | 🎉 -144 bytes |
| Date+Extensions.o | 47,304 bytes | 47,160 bytes | 🎉 -144 bytes |
| TokenizedControlView.o | 48,152 bytes | 48,008 bytes | 🎉 -144 bytes |
| String+Extension.o | 23,600 bytes | 23,456 bytes | 🎉 -144 bytes |
| DatePickerSelectionManager.o | 57,240 bytes | 57,096 bytes | 🎉 -144 bytes |
| FluentTheme+Tokens.o | 190,144 bytes | 189,992 bytes | 🎉 -152 bytes |
| TableViewHeaderFooterViewTokenSet.o | 87,000 bytes | 86,848 bytes | 🎉 -152 bytes |
| LabelTokenSet.o | 59,280 bytes | 59,128 bytes | 🎉 -152 bytes |
| BadgeFieldTokenSet.o | 55,096 bytes | 54,936 bytes | 🎉 -160 bytes |
| TableViewHeaderFooterView.o | 279,368 bytes | 279,208 bytes | 🎉 -160 bytes |
| TwoLineTitleView.o | 256,368 bytes | 256,208 bytes | 🎉 -160 bytes |
| PersonaBadgeViewDataSource.o | 37,432 bytes | 37,272 bytes | 🎉 -160 bytes |
| SideTabBarTokenSet.o | 57,104 bytes | 56,944 bytes | 🎉 -160 bytes |
| BadgeView.o | 629,968 bytes | 629,800 bytes | 🎉 -168 bytes |
| TabBarTokenSet.o | 47,640 bytes | 47,472 bytes | 🎉 -168 bytes |
| FluentListSection.o | 44,392 bytes | 44,224 bytes | 🎉 -168 bytes |
| TextFieldTokenSet.o | 81,992 bytes | 81,824 bytes | 🎉 -168 bytes |
| TooltipModifiers.o | 98,792 bytes | 98,624 bytes | 🎉 -168 bytes |
| CommandBarTokenSet.o | 71,112 bytes | 70,944 bytes | 🎉 -168 bytes |
| AvatarGroupTokenSet.o | 53,008 bytes | 52,840 bytes | 🎉 -168 bytes |
| PopupMenuSectionHeaderView.o | 24,944 bytes | 24,768 bytes | 🎉 -176 bytes |
| UIScrollView+Extensions.o | 34,016 bytes | 33,840 bytes | 🎉 -176 bytes |
| MSFAvatarPresence.o | 42,544 bytes | 42,368 bytes | 🎉 -176 bytes |
| FluentListSectionHeader.o | 47,624 bytes | 47,448 bytes | 🎉 -176 bytes |
| PopupMenuProtocols.o | 27,744 bytes | 27,568 bytes | 🎉 -176 bytes |
| IndeterminateProgressBarModifiers.o | 17,408 bytes | 17,232 bytes | 🎉 -176 bytes |
| AccessibleViewDelegate.o | 15,976 bytes | 15,800 bytes | 🎉 -176 bytes |
| ContentScrollViewTraits.o | 16,232 bytes | 16,056 bytes | 🎉 -176 bytes |
| PersonaButtonCarouselTokenSet.o | 41,304 bytes | 41,128 bytes | 🎉 -176 bytes |
| IndeterminateProgressBarTokenSet.o | 44,368 bytes | 44,192 bytes | 🎉 -176 bytes |
| UIApplication+Extensions.o | 12,264 bytes | 12,080 bytes | 🎉 -184 bytes |
| Calendar+Extensions.o | 43,112 bytes | 42,928 bytes | 🎉 -184 bytes |
| FluentTheme+visionOS.o | 11,248 bytes | 11,064 bytes | 🎉 -184 bytes |
| TooltipTokenSet.o | 57,736 bytes | 57,552 bytes | 🎉 -184 bytes |
| AvatarGroupModifiers.o | 11,256 bytes | 11,072 bytes | 🎉 -184 bytes |
| DayOfMonth.o | 58,856 bytes | 58,672 bytes | 🎉 -184 bytes |
| SwiftUI+ViewAnimation.o | 48,400 bytes | 48,216 bytes | 🎉 -184 bytes |
| TokenSet.o | 19,320 bytes | 19,136 bytes | 🎉 -184 bytes |
| CALayer+Extensions.o | 12,016 bytes | 11,832 bytes | 🎉 -184 bytes |
| GlobalTokens+UIKit.o | 19,456 bytes | 19,272 bytes | 🎉 -184 bytes |
| PeoplePickerTokenSet.o | 22,792 bytes | 22,608 bytes | 🎉 -184 bytes |
| PopupMenuTokenSet.o | 20,824 bytes | 20,640 bytes | 🎉 -184 bytes |
| FluentListSectionFooter.o | 34,784 bytes | 34,600 bytes | 🎉 -184 bytes |
| UIImage+Extensions.o | 12,048 bytes | 11,864 bytes | 🎉 -184 bytes |
| GenericDateTimePicker.o | 13,896 bytes | 13,712 bytes | 🎉 -184 bytes |
| GlobalTokens.o | 200,152 bytes | 199,968 bytes | 🎉 -184 bytes |
| UIKit+SwiftUI_interoperability.o | 12,304 bytes | 12,120 bytes | 🎉 -184 bytes |
| PersonaButtonCarouselModifiers.o | 11,296 bytes | 11,112 bytes | 🎉 -184 bytes |
| NotificationTokenSet.o | 94,024 bytes | 93,840 bytes | 🎉 -184 bytes |
| PersonaButtonTokenSet.o | 65,512 bytes | 65,328 bytes | 🎉 -184 bytes |
| SeparatorTokenSet.o | 39,672 bytes | 39,488 bytes | 🎉 -184 bytes |
| FluentTextInputError.o | 23,312 bytes | 23,128 bytes | 🎉 -184 bytes |
| ListItemModifiers.o | 26,200 bytes | 26,016 bytes | 🎉 -184 bytes |
| Compatibility.o | 11,928 bytes | 11,744 bytes | 🎉 -184 bytes |
| PersonaButtonModifiers.o | 11,256 bytes | 11,072 bytes | 🎉 -184 bytes |
| PopupMenuSection.o | 28,904 bytes | 28,720 bytes | 🎉 -184 bytes |
| EmptyTokenSet.o | 36,608 bytes | 36,424 bytes | 🎉 -184 bytes |
| DateComponents+Extensions.o | 12,224 bytes | 12,040 bytes | 🎉 -184 bytes |
| TokenizedControl.o | 14,096 bytes | 13,912 bytes | 🎉 -184 bytes |
| PopupMenuItemTokenSet.o | 31,656 bytes | 31,472 bytes | 🎉 -184 bytes |
| ShimmerTokenSet.o | 69,416 bytes | 69,232 bytes | 🎉 -184 bytes |
| ResizingHandleTokenSet.o | 44,136 bytes | 43,952 bytes | 🎉 -184 bytes |
| ContentHeightResolutionContext.o | 23,152 bytes | 22,968 bytes | 🎉 -184 bytes |
| Color+DynamicColor.o | 36,392 bytes | 36,208 bytes | 🎉 -184 bytes |
| ListActionItemModifiers.o | 17,120 bytes | 16,936 bytes | 🎉 -184 bytes |
| CommandingSection.o | 28,128 bytes | 27,944 bytes | 🎉 -184 bytes |
| Obscurable.o | 25,152 bytes | 24,960 bytes | 🎉 -192 bytes |
| SwiftUI+ViewPresentation.o | 58,416 bytes | 58,224 bytes | 🎉 -192 bytes |
| UIViewController+Navigation.o | 16,152 bytes | 15,960 bytes | 🎉 -192 bytes |
| ShapeCutout.o | 38,144 bytes | 37,944 bytes | 🎉 -200 bytes |
| ButtonTokenSet.o | 125,544 bytes | 125,344 bytes | 🎉 -200 bytes |
| TabBarItemTokenSet.o | 65,176 bytes | 64,976 bytes | 🎉 -200 bytes |
| AvatarTokenSet.o | 117,296 bytes | 117,080 bytes | 🎉 -216 bytes |
| BadgeViewTokenSet.o | 93,792 bytes | 93,576 bytes | 🎉 -216 bytes |
| ShyHeaderController.o | 253,872 bytes | 253,648 bytes | 🎉 -224 bytes |
| PillButtonTokenSet.o | 88,072 bytes | 87,848 bytes | 🎉 -224 bytes |
| BottomCommandingTokenSet.o | 72,520 bytes | 72,288 bytes | 🎉 -232 bytes |
| SegmentedControl.o | 333,640 bytes | 333,408 bytes | 🎉 -232 bytes |
| CardNudgeTokenSet.o | 86,576 bytes | 86,344 bytes | 🎉 -232 bytes |
| SegmentedControlTokenSet.o | 94,936 bytes | 94,632 bytes | 🎉 -304 bytes |
| NavigationBar.o | 554,696 bytes | 554,376 bytes | 🎉 -320 bytes |
| DateTimePickerView.o | 211,312 bytes | 210,936 bytes | 🎉 -376 bytes |
| CommandBar.o | 197,168 bytes | 196,776 bytes | 🎉 -392 bytes |
| ShimmerView.o | 198,912 bytes | 198,376 bytes | 🎉 -536 bytes |
| BottomSheetController.o | 525,512 bytes | 524,896 bytes | 🎉 -616 bytes |
| TooltipView.o | 204,096 bytes | 203,448 bytes | 🎉 -648 bytes |
| Avatar.o | 602,008 bytes | 601,344 bytes | 🎉 -664 bytes |
| DrawerController.o | 484,192 bytes | 483,512 bytes | 🎉 -680 bytes |
| FocusRingView.o | 842,384 bytes | 841,408 bytes | 🎉 -976 bytes |
| TableViewCell.o | 825,720 bytes | 824,736 bytes | 🎉 -984 bytes |
| MSFNotification.o | 139,808 bytes | 138,632 bytes | 🎉 -1,176 bytes |
| BottomCommandingController.o | 839,808 bytes | 838,176 bytes | 🎉 -1,632 bytes |
| MSFHeadsUpDisplay.o | 35,456 bytes | 33,736 bytes | 🎉 -1,720 bytes |
| MSFAvatarGroup.o | 32,152 bytes | 30,352 bytes | 🎉 -1,800 bytes |
| MSFPersonaButtonCarousel.o | 36,256 bytes | 34,456 bytes | 🎉 -1,800 bytes |
| MSFCardNudge.o | 39,544 bytes | 37,728 bytes | 🎉 -1,816 bytes |
| MSFAvatar.o | 30,672 bytes | 28,824 bytes | 🎉 -1,848 bytes |
| BadgeField.o | 575,552 bytes | 573,224 bytes | 🎉 -2,328 bytes |
| PopupMenuController.o | 378,064 bytes | 374,656 bytes | 🎉 -3,408 bytes |
| __.SYMDEF | 4,865,752 bytes | 4,847,920 bytes | 🎉 -17,832 bytes |
| FluentThemedHostingController.o | 63,544 bytes | 45,320 bytes | 🎉 -18,224 bytes |
| HUD.o | 266,200 bytes | 239,768 bytes | 🎉 -26,432 bytes |
</details>

### Verification

(how the change was tested, including both manual and automated tests)

<details>
<summary>Visual Verification</summary>

Tested a variety of components, including SwiftUI ones wrapped by FluentThemedHostingController.

Simulator Screen Recording - iPhone 15 Pro - 2024-06-07 at 10 28 30

Simulator Screen Recording - iPad Pro (11-inch) (4th generation) - 2024-06-07 at 13 36 07

</details>

### Pull request checklist

This PR has considered:
- [ ] Light and Dark appearances
- [x] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Different resolutions (1x, 2x, 3x)
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] iPad [Pointer interaction](https://developer.apple.com/documentation/uikit/pointer_interactions)
- [x] [SwiftUI](https://developer.apple.com/tutorials/swiftui) consumption (validation or new demo scenarios needed)
- [ ] Objective-C exposure (provide it only if needed)
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/fluentui-apple/pull/2042)

### CODE REVIEW

**Comment by harrieshin** [comment]
i expected https://github.com/microsoft/fluentui-apple/blob/main/Package.swift to be updated

**Comment by joannaquu** [comment]
`Removing legacy swizzling code for handling content insets in UIHostingController-backed UI. We can now use the public API sizingOptions = [.intrinsicContentSize].`

can you double check that avatar group is not affected? I remember there was a bug with our hosting view when we deprecated it

**Comment by joannaquu** [comment]
> `Removing legacy swizzling code for handling content insets in UIHostingController-backed UI. We can now use the public API sizingOptions = [.intrinsicContentSize].`
> 
> can you double check that avatar group is not affected? I remember there was a bug with our hosting view when we deprecated it

oh never mind `intrinsicContentSize` was the fix! #1831 for reference

File: ios/FluentUI/List/FluentList.swift [hunk 1]
```diff
@@ -44,12 +44,14 @@ public struct FluentList<ListContent: View>: View {
             case .insetGrouped:
                 list
                     .listStyle(.insetGrouped)
+                    .scrollContentBackground(.hidden)
```
Comment (joannaquu) [inline ios/FluentUI/List/FluentList.swift:47]: how come inset doesn't need?

**Review by joannaquu — APPROVED** [review]

File: ios/FluentUI/List/FluentList.swift (hunk 1 above)
Comment (mischreiber) [reply ios/FluentUI/List/FluentList.swift:47]: Inset doesn't seem to have a custom background color at the moment, so there's no need to remove the background.

@alexanderboswell, @nodes11, can either of you speak to why `.inset` exists but has no customization?

**Comment by mischreiber** [comment]
> i expected https://github.com/microsoft/fluentui-apple/blob/main/Package.swift to be updated

We really need to start testing this code via SPM one of these days :-/

**Comment by mischreiber** [comment]
> > `Removing legacy swizzling code for handling content insets in UIHostingController-backed UI. We can now use the public API sizingOptions = [.intrinsicContentSize].`
> > can you double check that avatar group is not affected? I remember there was a bug with our hosting view when we deprecated it
> 
> oh never mind `intrinsicContentSize` was the fix! #1831 for reference

Good call, `AvatarGroup` was one of the first things I checked too! I'll upload a gif of it working.

**Review by harrieshin — APPROVED** [review]

File: ios/FluentUI/List/FluentList.swift (hunk 1 above)
Comment (mischreiber) [reply ios/FluentUI/List/FluentList.swift:47]: Gonna merge for now but follow up offline :)

Comment (nodes11) [reply ios/FluentUI/List/FluentList.swift:47]: I'm not exactly sure. The styles we had an internal use case for were added and customizations were going to be added as needed.

**Pull request merged**